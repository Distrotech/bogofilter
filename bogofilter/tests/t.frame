# testing framework for /bin/sh

set -e

while : ; do
  TMPDIR=./checks.$$.`date +"%Y%m%dT%H%M%S"`
  mkdir $TMPDIR && break
  sleep 1
done

trap "if test x$SUPPRESS_DELETE = x ; then rm -r -f core ${TMPDIR} ; else echo remove ${TMPDIR} and possibly core yourself ; fi" 0
BOGOFILTER_DIR=$TMPDIR
export BOGOFILTER_DIR
../bogofilter -n </dev/null
