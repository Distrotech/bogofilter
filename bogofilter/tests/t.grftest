#!/bin/sh

. ${srcdir:=.}/t.frame

set -e

# Note:  This test DOES leave its output files on the hard drive.
#	 This makes it easier to see what has happened, which is
#		especially useful when something is different.
#
#        In directory t.systest.d are:
#
#		tests   - directory containing individual output files
#		grftest.out - comparison of directories:
#			      outputs - known correct outputs
#			      tests   - outputs of the individual tests

verbose=0
if [ "$1" = "-v" ]; then
    verbose=1
fi

[   -d ${TMPDIR} ] && rm -rf ${TMPDIR}
[ ! -d ${TMPDIR} ] && mkdir  ${TMPDIR}

# Graham config
cat /dev/null > ${TMPDIR}/cfg.g

# Robinson config
cat <<EOF > ${TMPDIR}/cfg.r
robx=0.415
min_dev=0.0
EOF

# Fisher config
cat <<EOF > ${TMPDIR}/cfg.f
robx=0.415
min_dev=0.1
ham_cutoff=0.0
EOF

# Fisher tristate
cat <<EOF > ${TMPDIR}/cfg.t
robx=0.415
min_dev=0.1
ham_cutoff=0.1
EOF

BOGOUTIL="../bogoutil"
BOGOFILTER="../bogofilter"
SYSTEST="${srcdir}/t.systest.d"

map_rc()
{
    (
	set +e
	eval "$@"
	a=$?
	[ $a -eq 0 ] && exit 0
	[ $a -eq 1 ] && exit 0
	echo exit $a
	exit $a
    )
}

run_test()
{
    id="$1"
    fil="$2"
    ver="$3"
    OUT="${TMPDIR}/tests/msg.$fil.$mth.$ver"
    OPTS="-$alg -t -$ver $CFG"
#echo $OPTS $OUT
    map_rc $BOGOFILTER $OPTS <${SYSTEST}/inputs/msg.$id.txt >${TMPDIR}/tests-$alg 
    sed s/,.version=.*// < ${TMPDIR}/tests-$alg > $OUT
#echo $BOGOFILTER $OPTS ${SYSTEST}/inputs/msg.$id.txt $OUT
}

[   -d ${TMPDIR}/tests ] && rm -f ${TMPDIR}/tests/*
[ ! -d ${TMPDIR}/tests ] && mkdir -p ${TMPDIR}/tests

for mth in g r f t ; do
    alg="$mth"
    [ "$alg" = "t" ] && alg=f
#   echo "***** a: $alg,  o: $mth *****"
    BOGOFILTER_DIR="${TMPDIR}/words.$mth"
    CFG="-c ${TMPDIR}/cfg.$mth"
    export BOGOFILTER_DIR
    [   -d $BOGOFILTER_DIR ] && rm -f $BOGOFILTER_DIR/*
    [ ! -d $BOGOFILTER_DIR ] && mkdir $BOGOFILTER_DIR
    $BOGOFILTER -$alg $CFG -s < ${SYSTEST}/inputs/spam.mbx
    $BOGOFILTER -$alg $CFG -n < ${SYSTEST}/inputs/good.mbx
#   ../bogoutil -w $BOGOFILTER_DIR .MSG_COUNT
    #
    # run tests for msg.[1-8].txt
    #
    for msg in ${SYSTEST}/inputs/msg.?.txt ; do
	tst=`echo $msg | sed s@${SYSTEST}/inputs/msg.@@ | sed s@.txt@@`
	args="$tst $tst v"
	run_test $args
    done

    rm -f ${TMPDIR}/tests-$alg
done

OUT="grftest.out"

printf "%2s     %-10s   %-10s   %-10s   %-10s\n" id graham robinson fisher tristate > ${TMPDIR}/tests/$OUT
for out in ${TMPDIR}/tests/msg.?.r.v ; do
    id=`echo $out | sed s@${TMPDIR}/tests/msg.@@ | sed s@.r.v@@`
    g=`cat ${TMPDIR}/tests/msg.$id.g.v | tr "NYU" "hSu"`
    r=`cat ${TMPDIR}/tests/msg.$id.r.v | tr "NYU" "hSu"`
    f=`cat ${TMPDIR}/tests/msg.$id.f.v | tr "NYU" "hSu"`
    t=`cat ${TMPDIR}/tests/msg.$id.t.v | tr "NYU" "hSu"`
    printf "%2s   %c %s   %c %s   %c %s   %c %s\n" $id $g $r $f $t >> ${TMPDIR}/tests/$OUT
done

OUTPUTS=${SYSTEST}/outputs
if  [ $verbose -eq 0 ]; then
    cmp ${OUTPUTS}/$OUT ${TMPDIR}/tests/$OUT
else
    diff -u -s --brief ${OUTPUTS}/$OUT ${TMPDIR}/tests/$OUT
fi
