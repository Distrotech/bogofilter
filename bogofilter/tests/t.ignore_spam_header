#! /bin/sh

. ${srcdir=.}/t.frame

# test description here
#
# test below
# remember to use ${srcdir}

shn=`./spam_header_name`
hdr="From somewhere@example.com Fri Nov 22 11:30:23 2002"
shn1="$shn: blah1"
shn2="$shn: blah2"

( echo $hdr ; echo $shn1 ; echo ; echo $shn2 ) >$TMPDIR/input
tail -2 $TMPDIR/input >$TMPDIR/want
$VAL ../bogofilter -p -e -C < $TMPDIR/input | sed -n '/^$/,$ p' > $TMPDIR/out

cmp $TMPDIR/out $TMPDIR/want
