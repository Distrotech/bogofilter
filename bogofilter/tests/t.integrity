#! /bin/sh

. ${srcdir}/t.frame
# test description here
#
# test below
# remember to use ${srcdir}
( cd ${TMPDIR} && uudecode ../t.integrity.uu )
../bogofilter -e -p <${TMPDIR}/input >${TMPDIR}/output
head -1 ${TMPDIR}/output | cmp - ${TMPDIR}/input
