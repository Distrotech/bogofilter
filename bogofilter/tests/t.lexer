#! /bin/sh

. ${srcdir:=.}/t.frame

set -e

cat inputs/t.lexer.inp | ../bogolexer -p | sort -u > ${TMPDIR}/t.lexer.out

cmp outputs/t.lexer.out ${TMPDIR}/t.lexer.out
