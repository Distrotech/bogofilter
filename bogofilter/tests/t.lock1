#! /bin/sh

. ${srcdir:=.}/t.frame

# this test checks that bogofilter -u works
../bogofilter -u -p -C -e -d $TMPDIR </dev/null >/dev/null
