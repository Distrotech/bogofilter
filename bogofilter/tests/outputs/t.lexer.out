account
all
and
any
anytime
back
behind
biggest
button
cancelable
cancellation
change
character
city
city's
clancy
clancy's
click
complete
conditions
controversial
credits
crisis
dvds
for
free
from 
giuliani
guided
handling
his
history
inc
including
instructions
jack
junkflix
king's
latest
leadership
learned
length
lessons
mayor
month-to-month
monthly
most
netflix
new
nobody
nowhere.com
office
partial
period
periods
please
popular
program
rabbit
red
reflects
refunds
rent
reserves
right
rudolph
ryan
shipping
standard
stephen
story
subject
subscription
terms
the
there
thriller
through
time
tom
trial
visit
want
who
with
www.netflix.com
york
you
your
